File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/FizzBuzzListCommand.java
package fizzbuzz;

import fizzbuzz.domain.model.FizzBuzzList;
import fizzbuzz.domain.model.FizzBuzzValue;
import fizzbuzz.domain.type.FizzBuzzType;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class FizzBuzzListCommand implements IFizzBuzzCommand {
  private FizzBuzzType _type;
  private FizzBuzzList _list;

  public FizzBuzzListCommand(FizzBuzzType type) {
    _type = type;
  }

  @Override
  public IFizzBuzz execute(int maxNumber) {
    if (maxNumber < 0)
    throw new IllegalArgumentException(
      "FizzBuzzList can't generate by minus number"
    );
    List<FizzBuzzValue> list = new ArrayList<>();
    _list = new FizzBuzzList(list);

    IntStream.rangeClosed(1, maxNumber).forEach(
      number -> _list = _list.add(_type.generate(number))
    );

    return _list;
  }

}

File: dev/20190902/k2works/app/FizzBuzz/Domain/FizzBuzzListCommand.cs
using System;
using System.Linq;

namespace FizzBuzz.Domain
{
    public class FizzBuzzListCommand: IFizzBuzzCommand
    {
        private FizzBuzzType _type;

        public FizzBuzzListCommand(FizzBuzzType type)
        {
            _type = type;
        }

        public IFizzBuzz Execute(int number)
        {
            if (number < 0) throw new ArgumentException("FizzBuzzList can't generate by minus number'");
            return new FizzBuzzList(Enumerable.Range(1, number).Select(i => _type.Generate(i)).ToList());
        }
    }
}